docs/source/upgrade/example/coinV2/daml/UpgradeFromCoinV1.daml

```diff
@@ -1,26 +1,11 @@
 -- Copyright (c) 2020 The DAML Authors. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
 
-module Upgrade where
-
--- COIN_BEGIN
-template Coin
-  with
-    issuer : Party
-    owner : Party
-  where
-    signatory issuer, owner
--- COIN_END
-
--- COIN_AMOUNT_BEGIN
-template CoinWithAmount
-  with
-    issuer : Party
-    owner : Party
-    amount : Int
-  where
-    signatory issuer, owner
--- COIN_AMOUNT_END
+-- UPGRADE_MODULE_BEGIN
```

Comment:
I’d keep it like this for now, but we might want to consider splitting this up into 3 packages,

1. version 1
2. version 2
3. upgrades from version 1 to version 2

Reply:
That is indeed how davl lays things out too.

Reply:
what does having 3 packages buy us?

Reply:
Mostly it’s about separating concerns, i.e., don’t entangle the new state with the way of getting to that state. However there is one advantage that becomes relevant once your project reaches a certain age (i.e. number of upgrades):

Since a a DAR is always self-contained, keeping the upgrades and the new models together means that your DAR will get bigger and bigger since you can never drop old versions even if you write something that only references the new models, e.g., a trigger that only cares about the models of a specific version. If you keep it separate on the other hand, your trigger can only depend on the new models.